cross-modal-search/README.md

```diff
@@ -70,128 +58,146 @@ for encoding image and text respectively.
+Once you run this command, you should see a JSON output returned to you. This contains the five most semantically similar images sentences to the text input you provided in the `data` parameter.
+Note, that the toy-data only contains two images.
+Feel free to alter the text in the 'data' parameter and play around with other queries (this is only fun with the large dataset)! For a better understanding of the parameters see the table below. 
+|  |  |
+|--|--|
+| `top_k` | Integer determining the number of sentences to return |
+| `mode` | Mode to trigger in the call. See [here](https://docs.jina.ai/chapters/rest/) for more details |
+| `data` | Text input to query |
+ 
+#### Using Jina Box; our frontend search interface
 
+**Jina Box** is a light-weight, highly customizable JavaScript based front-end search interface. To use it for this example, begin by opening the REST API interface. 
 
-## Use Docker image from the jina hub
+```sh
+python app.py -t query_restful
+```
 
+In your browser, open up the hosted Jina Box on [jina.ai/jinabox.js](https://jina.ai/jinabox.js/). In the configuration bar on the left-hand side, choose a custom endpoint and enter the following: `http://127.0.0.1:45678/search`.
+You can type search queries into the text box on the right-hand side!
+
+______
+
+## 🌀 Flow diagram
+This diagram provides a visual representation of the Flows in this example; Showing which executors are used in which order.
+Remember, our goal is to compare vectors representing the semantics of images with vectors encoding the semantics of short text descriptions.
+
+### Indexing
+![](visualizations/index-flow.png)
+As you can see, the Flow that indexes the data contains three parallel branches. 
+The upper branch contains only a key-value indexer for the images that we use as a lookup (like a Dictionary in Python).
+The middle branch contains the transformations to get from the JPG to the image vector, and the lower branch takes care of turning 
+text descriptions into vectors. To have low latency at query time, we store the computed vectors on disk.
+   
+![](visualizations/query-flow.png)
+This Flow shows what happens when a user queries our data. First, the provided text description is passed through the text Encoder which turns
+it into a vector. Now, we use our image vector index to find the most similar image encodings to the previously computed text vector from the user.
+Because the user does not want to see the vector as a result, but the image this vector belongs to we use the key-value lookup to get from image vector to human-interpretable JPG image.
+Note, that this Flow only shows how to search for images using text. The example actually support searching for text using images as well. 
+As an exercise, you can think of the required steps for that and check against our [Flow configuration](flows/flow-query.yml).
+
+## 🔮 Overview of the files
+
+*Add a list with all folders/files in the example:*
```

Comment:
Could you remove this sentence please?

Reply:
984b5ae21733967c6d38c821527ca50356f672cb